### PR TITLE
Bls benchmarking and optimisation

### PR DESCRIPTION
**Pull request by mpetrunic**
- benchmark suites for bls
- optimisation of aggregated signatures

Current suites are not really good as sample setups takes too much time, I will open issue to generate json sample cases once and always run against those sample cases.

### CODE REVIEW

**Review by GregTheGreek — DISMISSED** [review]

File: packages/bls/src/helpers/g1point.ts [hunk 1]
```diff
@@ -28,6 +28,10 @@ export class G1point {
     return new G1point(sum);
   }
 
+  public addRaW(other: bytes48): G1point {
```
Comment (wemeetagain) [inline packages/bls/src/helpers/g1point.ts]: maybe `addRaw`

File: packages/bls/src/index.ts [hunk 2]
```diff
@@ -98,20 +96,50 @@ export function verifyMultiple(publicKeys: BLSPubkey[], messageHashes: bytes32[]
...
+      ElipticCurvePairing.pair(
+        keyFinal,
+        g2Final
+      )
+    );
+
+    const e2 = ElipticCurvePairing.pair(G1point.generator(), sig);
     return e2.equals(eCombined);
   } catch (e) {
+    console.log(e);
```
Comment (wemeetagain) [inline packages/bls/src/index.ts]: should this be there?

**Review by wemeetagain — CHANGES_REQUESTED** [review]

File: packages/bls/src/index.ts (hunk 2 above)
Comment (mpetrunic) [reply packages/bls/src/index.ts]: probably not, accident

**Comment by mpetrunic** [comment]
@GregTheGreek @wemeetagain can we merge this?

**Review by GregTheGreek — APPROVED** [review]

**Comment by mpetrunic** [comment]
@wemeetagain this is kinda important

**Review by wemeetagain — APPROVED** [review]

**Pull request merged**